homeassistant/components/onkyo/media_player.py

```diff
@@ -511,9 +511,9 @@ def update(self) -> None:
         elif ATTR_PRESET in self._attr_extra_state_attributes:
             del self._attr_extra_state_attributes[ATTR_PRESET]
         if self._supports_volume:
-            # AMP_VOL/MAX_RECEIVER_VOL*(MAX_VOL/100)
-            self._attr_volume_level = (
-                volume_raw[1] / self._receiver_max_volume * (self._max_volume / 100)
+            # AMP_VOL / (MAX_RECEIVER_VOL * (MAX_VOL / 100))
```

Comment:
We should have the equation be consistent in the code and the comment.  Right now the comment has an extra set of parenthesis and the code does not.  While they both evaluate the same, let's be explicit.

Reply:
Parenthesis are useful in the comment to show grouping of related terms.
In the linked PR https://github.com/home-assistant/core/pull/42236 @MartinHjelmare suggested to remove them from code, so I kept it that way.

Reply:
Because of the above I will close this discussion. If there is strong support for adding back the parenthesis from the other reviewers, I will do it.

Reply:
Without knowing the data types involved it's not clear if they are equivalent.  The good news is the code should work correctly regardless of the data types.

In the comment we divide MAX_VOLUME / 100 first which will result in an integer if both parameters are integers.  If max volume is 49 that will yield 0; if max volume is 150 it'll yield 1. Which is not what you want.

In the code the two multiplications will happen first then the division so I think it'll all work out and you'll get the right answer.

Reply:
> In the comment we divide MAX_VOLUME / 100 first which will result in an integer if both parameters are integers. If max volume is 49 that will yield 0; if max volume is 150 it'll yield 1.

That's not true.

```
Python 3.12.4 (main, Jun  8 2024, 18:29:57) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.22.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: MAX_VOLUME = 49

In [2]: type(MAX_VOLUME)
Out[2]: int

In [3]: test = MAX_VOLUME / 100

In [4]: type(test)
Out[4]: float

In [5]: test
Out[5]: 0.49

In [6]: MAX_VOLUME = 150

In [7]: type(MAX_VOLUME)
Out[7]: int

In [8]: test = MAX_VOLUME / 100

In [9]: type(test)
Out[9]: float

In [10]: test
Out[10]: 1.5
```

Reply:
In Python dividing two ints using / gives a float. If you want integer division, you need to use //.

Reply:
Learned something new.  Thank you.